en/docs/cookies/global_tensor.md

```diff
@@ -0,0 +1,513 @@
+
+Because the tensor adopts the Multi-Client mode, we need to launch different processes corresponding to each device. In OneFlow, all processes need to launch the same scripts, and different processes distinguish process ID and establish communications according to different environment variables.
+
+环境变量说明：
+
+Notes of environment variables:
+
+- `MASTER_ADDR`：多机训练的第 0 号机器的 IP；
+- `MASTER_PORT`：多机训练的第 0 号机器的监听端口，不与已经占用的端口冲突即可；
+- `WORLD_SIZE`：整个集群中计算设备的数目，因为目前还不支持各个机器上显卡数目不一致，因此 `WORLD_SIZE` 的数目实际上是 $机器数目 \times 每台机器上的显卡数目$。[创建 Global Tensor](#global-tensor_2) 中的示例是单机 2 卡的情况，因此 `WORLD_SIZE=2`；
+- `RANK`：集群内所有机器下的进程编号；
+- `LOCAL_RANK`：单个机器内的进程编号；
+
+- `MASTER_ADDR`：the IP of Machine 0 that is trained on multiple machines;
+- `MASTER_PORT`：the monitor port of Machine 0 that is trained on multiple machines, and this port shouldn’t conflict with the occupied ports;
+- `WORLD_SIZE`: the number of computing device in the whole cluster. Because it’s still not feasible to configure different number of GPUs on each device, the `WORLD_SIZE` equals the machine numbers multiplies the GPU numbers on each machine. In the previous case, we [create the global tensor](#global-tensor_2) in single machine 2 GPUs environment, so the `WORLD_SIZE=2`;
+- `RANK`：the process ID of all devices in the whole cluster;
+- `LOCAL_RANK`：the process ID of single device;
+
+
+`RANK` 和 `LOCAL_RANK` 的区别：
+
+Differences between `RANK` and `LOCAL_RANK`: 
+
+- 当是单机训练（单机单卡或单机多卡）时，两者相等；
+- 当是多机训练时，每台机器上的 `LOCAL_RANK` 的上限，就是每台机器上的计算设备的数目；`RANK` 的上限，就是所有机器上所有计算设备的总和，它们的编号均从0开始。（因为编号从0开始，所以不包含上限）。
+
+- For single machine training, including single-machine single-GPU and single-machine multi-GPU, `RANK` equals to `LOCAL_RANK`;
+- For multi-machine training, the upper limit to `LOCAL_RANK` for each device is the number of computing devices on each machine; the upper limit to `RANK` is the sum of computing devices on all machines, and all devices are numbered from 0. (Because these computing devices are numbered from 0, the upper limit doesn’t exist.)
+
+以 `2 机 2 卡` 为例，每张显卡的 `LOCAL_RANK` 与 `RANK` 对应情况如下：
+
+Take `2 Machines 2 GPUs` for example, the corresponding relationship between `LOCAL_RANK` and `RANK` for each GPU is listed as follows:
+
+
+|                      | RANK | LOCAL_RANK |
+| -------------------- | ---- | ---------- |
+| 机器 0 的第 0 张显卡 | 0    | 0          |
+| 机器 0 的第 1 张显卡 | 1    | 1          |
+| 机器 1 的第 0 张显卡 | 2    | 0          |
+| 机器 1 的第 1 张显卡 | 3    | 1          |
+
+|               | RANK | LOCAL_RANK |
+| ------------- | ---- | ---------- |
+| GPU 0 on Machine 0 | 0    | 0          |
+| GPU 0 on Machine 1 | 1    | 1          |
+| GPU 1 on Machine 0 | 2    | 0          |
+| GPU 1 on Machine 1 | 3    | 1          |
+
+使用环境变量启动虽然繁琐，但是适用性广，可以采用任意的方式来启动进程。
+
+Although it is complicated to utilize environment variables launching, this approach is widely applicable because users can adopt random ways to launch the process.
```

Comment:
```suggestion
Although it is complicated to utilize environment variables launching, this approach is widely applicable because users can adopt random ways to launch the processes.
```